### PR TITLE
fix pricing table and coupon input

### LINKED ISSUE #556: The free plan price is crossed out when the promotion is active
**Is your feature request related to a problem? Please describe.**
When the promotion is applied for all plans, `it looks like` it's also applied to the free plan:
image

We should not consider it as a discounted plan.

**Describe the solution you'd like**
It should just stay as 0$ 

**Describe alternatives you've considered**
We could even remove this plan and replace it with a simple button `downgrade to free plan`

### LINKED ISSUE #557: Coupon code is automatically applied to all plans in the UI, even to those that are not discounted
**Is your feature request related to a problem? Please describe.**
When the promo is applied only to the specific plan, the coupon code and the label `20% off with spring2021` is visible all the time:
image
Because of that, we're not able to apply another coupon to different, not discounted plans.

**Describe the solution you'd like**
I think that the coupon + label should be added only when the chosen plan is discounted in the promo. We should be also able to edit this field and apply different coupon

### PR DESCRIPTION
**Pull request by mostafa-hisham**
fixes #557
fixes #556

### CODE REVIEW

File: assets/js/pricing-modal.js [hunk 1]
```diff
@@ -1060,13 +1115,24 @@
 	 */
 	$(d).on('modalClosed.imagify', '.imagify-payment-modal', function () {
 		// Reset viewing class & aria-labelledby.
+		$('#imagify-coupon-code').val('');
+
+		/*$('.imagify-pre-checkout-offers .imagify-offer-onetime .imagify-offer-content').html('');
+		$('.imagify-pre-checkout-offers .imagify-offer-monthly .imagify-offer-content').html('');*/
```
Comment (iCaspar) [inline assets/js/pricing-modal.js]: Remove Commented-out code.

File: assets/js/pricing-modal.js [hunk 2]
```diff
@@ -568,7 +568,15 @@
 								ot_html += $tpl[0].outerHTML;
 							});
 						}
-
+						if (promo_datas){
+							var appliseTo = imagifyModal.getPromoAppliesTo(promo_datas);
+							if ( appliseTo.includes(suggested.mo.plan_label) || appliseTo.includes(suggested.ot.plan_label) ){
+								$('#imagify-coupon-code').val(promo_datas.label);
```
Comment (iCaspar) [inline assets/js/pricing-modal.js]: Type-o in variable name - appliesTo

File: assets/js/pricing-modal.js [hunk 3]
```diff
@@ -1176,6 +1251,18 @@
 			$target_line.find('.imagify-inline-options').find('input:radio:checked').trigger('change.imagify');
 		}
 
+		$coupon_input = $('#imagify-coupon-code');
+		$coupon_input.val('');
+
+		if ( w.imagify_discount_datas ){
+			var appliseTo = imagifyModal.getPromoAppliesTo(w.imagify_discount_datas);
+			if ( appliseTo.includes(datas[Object.keys(datas)[0]].label) ){
```
Comment (iCaspar) [inline assets/js/pricing-modal.js]: Type-o same as above -- appliesTo

**Review by iCaspar — CHANGES_REQUESTED** [review]
A couple very small things.

File: assets/js/pricing-modal.js (hunk 1 above)
Comment (mostafa-hisham) [reply assets/js/pricing-modal.js]: done

File: assets/js/pricing-modal.js (hunk 3 above)
Comment (mostafa-hisham) [reply assets/js/pricing-modal.js]: changed to `applies_to`

File: assets/js/pricing-modal.js (hunk 2 above)
Comment (mostafa-hisham) [reply assets/js/pricing-modal.js]: changed to `applies_to`

**Review by iCaspar — APPROVED** [review]

**Comment by iCaspar** [comment]
Ready for @wp-media/qa

**Comment by piotrbak** [comment]
@mostafa-hisham When promo is set to all plans, the coupon is not applied at all, but the prices are crossed out:
image

The discount text is also not present at the end of the checkout process:
image

The 0 is crossed out:
image
image

**Comment by mostafa-hisham** [comment]
@piotrbak 

> @mostafa-hisham When the promo is set to all plans, the coupon is not applied at all, but the prices are crossed out:
> image
> 
**FIXED**

> The discount text is also not present at the end of the checkout process:
> image
> 

this part is coming from "https://app.tst.imagify.io/#/plugin/0/0/15/wpr10/" in <iframe>
I think it is related to this issue https://github.com/wp-media/imagify/issues/1580

> The 0 is crossed out:
> image
> image
>
**FIXED**

**Comment by piotrbak** [comment]
@mostafa-hisham Now it's possible to apply a promo coupon to the plan that's not in promotion. To reproduce:
1. Promo is set to have only unlimited in promotion
2. Go to checkout, choose Growth
3. Add the promo coupon (`spring`) and click ok
4. The coupon is applied
5. Go to checkout, the checkout is not possible because the coupon is not valid.

image

**Comment by mostafa-hisham** [comment]
@piotrbak 
please check if you can apply a promo coupon to the plan that's not in the promotion
after this change

**Review by piotrbak — APPROVED** [review]
Seems that we're fine there. The PR wasn't tested against a situation when there are two plans in the promo campaign. John wasn't able to set it up.
[testrail-report-218.pdf](https://github.com/wp-media/imagify-plugin/files/6823141/testrail-report-218.pdf)

**Pull request merged**